cEP-0025.md

```diff
@@ -0,0 +1,116 @@
+# Integration of pyflakes-enhanced-AST into coala
+
+| Metadata |                                                 |
+| -------- | ----------------------------------------------- |
+| cEP      | 22                                              |
+| Version  | 1.0                                             |
+| Title    | Integration of pyflakes-enhanced-AST into coala |
+| Authors  | Ankit Joshi <mailto:ajankit2304@gmail.com>      |
+| Status   | Proposed                                        |
+| Type     | Feature                                         |
+
+## Abstract
+
+This document describes how to create a metabear to integrate
+pyflakes-enhanced-AST into coala.
+
+## Introduction
+
+The current approach of plugin development for flake8 does not utilises
+the full potential of enhanced AST rather they simply depend on python AST
+and so, a whole lot of rework is required to do the basic traversing
+and collection of important nodes.
+
+Pyflakes provides with a basic API that does the traversing.
+So, if a developer uses enhanced AST he just needs to work on
+the implementation of the new logic that his/her plugin provides
+and not about the fidelity of the basic node handlers.
+
+## Proposed Change
+
+Here is the detailed implementation stepwise:
+
+1. There will be a separate repository named as coala-pyflakes which will
+   be installable via `pip install coala-pyflakes`.
+2. The repository will contain a new package `pyflakes-plugins` which
```

Comment:
If this package contains primarily *bears*, it needs to include either *bears* or *coala* in the name.

But it might be worth creating a non-coala API for these linter plugins - an API which is similar to other frameworks like flake8, so they could also run the same linters.

Then the package contains a bunch of generic pyflakes plugins, a coala meta bear (the pyflakes ast provider), and a coala bear meta wrapper (which creates a wrapper bear for each of the plugins).

Or we could have two packages, one for the generic pyflakes plugins, and a different one for the coala glue.